applicationset/status/resource_status.go

```diff
@@ -7,31 +7,17 @@ import (
 func BuildResourceStatus(statusMap map[string]argov1alpha1.ResourceStatus, apps []argov1alpha1.Application) map[string]argov1alpha1.ResourceStatus {
 	appMap := map[string]argov1alpha1.Application{}
 	for _, app := range apps {
-		appCopy := app
 		appMap[app.Name] = app
 
 		gvk := app.GroupVersionKind()
-		// Create status if it does not exist
```

Comment:
i can revert this but i think its worth removing. when reviewing the status code, this threw me through a loop.

Reply:
I couldn't figure out why we are looping over the map to grab the cached status and overwrite everything on line 28. I suppose we potentially save a few bytes by not recreating the status struct but I think the readability is worth the trade-off.